component/client/ClientChat.jsx

```diff
@@ -1,14 +1,32 @@
 import React, {Component} from "react";
 import {Panel} from "react-bootstrap";
 import ClientMessenger from "./ClientMessenger.jsx";
+import RoomService from "./RoomService";
 
 class ClientChat extends Component {
+
+  room: RoomService;
+
   constructor(props) {
     super(props);
+    this.state = {
+      history: []
+    };
+    // set up Room Service
+    this.room = new RoomService(props.url, (message) => {
```

Comment:
for posteriority: when a Chrome browser action is closed, all state is lost including the state needed by firebase to maintain its singleton property. Therefore, to avoid the cost of constant reinstantiation of the firebase database, we should figure out how to connect this with the background.js and communicate with it via the popup.html/js